util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -672,6 +672,34 @@ object AsyncStream {
       case Some(a) => of(a)
     }
 
+  /**
+   * Like `flatMap` but can be arbitrarily nested.
+   *
+   * The 19.7.0 release introduces an optimization that makes `Future#map`
+   * bypass the scheduler and run immediately, which means the `flatMap`
+   * implementation for `AsyncStream` is no longer stack-safe. The
+   * implementation here uses `flatMap` on futures to avoid the issue and
+   * maintain stack safety.
+   */
+  private def stackSafeFlatMap[A, B](as: AsyncStream[A])(f: A => AsyncStream[B]): AsyncStream[B] =
+    as match {
+      case Empty => empty
+      case FromFuture(fa) => Embed(fa.flatMap(a => Future.value(f(a))))
+      case Cons(fa, more) =>
+        Embed(fa.flatMap(a => Future.value(f(a)))) ++ stackSafeFlatMap(more())(f)
+      case Embed(fas) =>
+        Embed(fas.flatMap(as => Future.value(stackSafeFlatMap(as)(f))))
+    }
```

Comment:
Was chatting with @travisbrown on Gitter and I think it's worth considering fixing flatMap (i.e. just replace flatMap with the implementation of stackSafeFlatMap above).

The performance gains of bypass scheduler apply to Future, but they weren't intended to break stack-safety of AsyncStream.flatMap. `flatMap` is commonly used for recursive constructs, we should make it safe to use that way.

Reply:
@luciferous Updated the PR, thanks!